### PR TITLE
FeedPackageIdentity comparisons should be case insensitive

### PR DESCRIPTION
**Pull request by scottbommarito**
`abcd 1.0.0-abc` should be equal to `ABCD 1.0.0-ABC`

(this is a fix for the V3 monitoring job)

### CODE REVIEW

File: src/Catalog/Helpers/FeedPackageIdentity.cs [hunk 1]
```diff
@@ -30,12 +30,12 @@ public FeedPackageIdentity(PackageIdentity package)
 
         public bool Equals(FeedPackageIdentity other)
         {
-            return Id == other.Id && Version == other.Version;
+            return Id.ToLowerInvariant() == other.Id.ToLowerInvariant() && Version.ToLowerInvariant() == other.Version.ToLowerInvariant();
```
Comment (skofman1) [inline src/Catalog/Helpers/FeedPackageIdentity.cs]: please add UTs that make sure Equals and GetHashCode are equivalent. I.e. if Equals returns true for two FeePackageIdentity classes, their hash codes are also identical (and vice versa).

Comment (skofman1) [inline src/Catalog/Helpers/FeedPackageIdentity.cs]: please add UTs that make sure Equals and GetHashCode are equivalent. I.e. if Equals returns true for two FeePackageIdentity classes, their hash codes are also identical (and vice versa).

**Comment by scottbommarito** [comment]
@skofman1 added tests as requested

**Comment by scottbommarito** [comment]
@skofman1 added more tests as requested offline

**Pull request merged**